Makefile

```diff
@@ -85,6 +85,47 @@ GO_CMD_PATHS := $(filter-out library, $(wildcard cmd/*))
 GO_CMD_NAMES := $(notdir $(GO_CMD_PATHS))
 GO_CMD_BUILDS := $(addprefix build/bin/, $(GO_CMD_NAMES))
 
+
+
+# Our custom config is located in nix/nix.conf
+export NIX_USER_CONF_FILES = $(PWD)/nix/nix.conf
+# Location of symlinks to derivations that should not be garbage collected
+export _NIX_GCROOTS = /nix/var/nix/gcroots/per-user/$(USER)/status-go
+
+
+# MacOS root is read-only, read nix/README.md for details
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+export NIX_IGNORE_SYMLINK_STORE=1
+endif
+#----------------
+# Nix targets
+#----------------
+
+
+SHELL := ./nix/scripts/shell.sh
+
+shell: export TARGET ?= default
+shell: ##@prepare Enter into a pre-configured shell
```

Comment:
This file doesn't have the script that provides help message from `make` so those are just comments.